code/game/machinery/suit_storage_unit.dm

```diff
@@ -1,184 +1,406 @@
+	helmet_type  = /obj/item/clothing/head/helmet/space/hardsuit/ert/engineer
+	mask_type    = /obj/item/clothing/mask/breath
+	storage_type = /obj/item/tank/emergency_oxygen/double
+
+/obj/machinery/suit_storage_unit/ert/medical
+	suit_type    = /obj/item/clothing/suit/space/hardsuit/ert/medical
+	helmet_type  = /obj/item/clothing/head/helmet/space/hardsuit/ert/medical
+	mask_type    = /obj/item/clothing/mask/breath
+	storage_type = /obj/item/tank/emergency_oxygen/double
+
+
+
+/obj/machinery/suit_storage_unit/Initialize()
+	. = ..()
+	if(suit_type)
+		suit = new suit_type(src)
+	if(helmet_type)
+		helmet = new helmet_type(src)
+	if(mask_type)
+		mask = new mask_type(src)
+	if(storage_type)
+		storage = new storage_type(src)
+	update_icon()
+
+	//move this into machinery eventually...
+	if(occupant_typecache)
+		occupant_typecache = typecacheof(occupant_typecache)
+
+/obj/machinery/suit_storage_unit/Destroy()
+	QDEL_NULL(suit)
+	QDEL_NULL(helmet)
+	QDEL_NULL(mask)
+	QDEL_NULL(storage)
+	return ..()
 
 /obj/machinery/suit_storage_unit/update_icon()
-	var/hashelmet = 0
-	var/hassuit = 0
-	var/hashuman = 0
-	if(HELMET)
-		hashelmet = 1
-	if(SUIT)
-		hassuit = 1
-	if(OCCUPANT)
-		hashuman = 1
-	icon_state = text("suitstorage[][][][][][][][][]",hashelmet,hassuit,hashuman,src.isopen,src.islocked,src.isUV,src.ispowered,src.isbroken,src.issuperUV)
+	cut_overlays()
 
+	if(uv)
+		if(uv_super)
+			add_overlay("super")
+		else if(occupant)
+			add_overlay("uvhuman")
+		else
+			add_overlay("uv")
+	else if(state_open)
+		if(stat & BROKEN)
+			add_overlay("broken")
+		else
+			add_overlay("open")
+			if(suit)
+				add_overlay("suit")
+			if(helmet)
+				add_overlay("helm")
+			if(storage)
+				add_overlay("storage")
+	else if(occupant)
+		add_overlay("human")
+
+/obj/machinery/suit_storage_unit/attackby(obj/item/I as obj, mob/user as mob, params)
+	if(!is_operational())
+		return
+	if(isscrewdriver(I))
+		panel_open = !panel_open
+		playsound(loc, I.usesound, 100, 1)
+		to_chat(user, text("<span class='notice'>You [panel_open ? "open up" : "close"] the unit's maintenance panel.</span>"))
+		updateUsrDialog()
 
 /obj/machinery/suit_storage_unit/power_change()
-	if( powered() )
-		src.ispowered = 1
-		stat &= ~NOPOWER
-		src.update_icon()
+	..()
+	if(!is_operational() && state_open)
+		open_machine()
+		dump_contents()
+	update_icon()
+
+
+/obj/machinery/suit_storage_unit/proc/dump_contents()
+	dropContents()
+	helmet = null
+	suit = null
+	mask = null
+	storage = null
+	occupant = null
+
+/obj/machinery/suit_storage_unit/deconstruct(disassembled = TRUE)
+	open_machine()
+	dump_contents()
+	new /obj/item/stack/sheet/metal (loc, 2)
+	qdel(src)
+
+/obj/machinery/suit_storage_unit/MouseDrop_T(atom/A, mob/user)
+	if(user.stat || user.lying || !Adjacent(user) || !Adjacent(A) || !isliving(A))
+		return
+	var/mob/living/target = A
+	if(!state_open)
+		to_chat(user, "<span class='warning'>The unit's doors are shut!</span>")

```

Comment:
I'd refer to all the following `unit's` as `[src]'s`

Reply:
^

Reply:
seems unnecessary to be saying `The suit storage unit's doors are shut!` rather than `The unit's doors are shut!` - as this only occurs when you're actively trying to interact with it.

If it was a wider message to everyone, then yes, it'd make sense to specify.

If you require it changed, then fine, I shall do so - but I think it is the worse option.

Reply:
I prefer it be referred to by [src] instead of unit, as it also accounts for suit storage unit with different name, sprite, or purpose.